app/src/main/java/org/love2d/android/LoveDocumentsProvider.java

```diff
@@ -0,0 +1,713 @@
+        } else {
+            path = path.substring(rootPath.length() + 1);
+        }
+
+        return ROOT + ':' + path;
+    }
+
+    /**
+     * Add a representation of a file to a cursor.
+     *
+     * @param result the cursor to modify
+     * @param docId  the document ID representing the desired file (may be null if given file)
+     * @param file   the File object representing the desired file (may be null if given docID)
+     * @throws FileNotFoundException
+     */
+    private void includeFile(MatrixCursor result, String docId, File file)
+            throws FileNotFoundException {
+        if (docId == null) {
+            docId = getDocIdForFile(file);
+        } else {
+            file = getFileForDocId(docId);
+        }
+
+        int flags = 0;
+
+        if (file.isDirectory()) {
+            // Request the folder to lay out as a grid rather than a list. This also allows a larger
+            // thumbnail to be displayed for each image.
+            //            flags |= Document.FLAG_DIR_PREFERS_GRID;
+
+            // Add FLAG_DIR_SUPPORTS_CREATE if the file is a writable directory.
+            if (file.isDirectory() && file.canWrite()) {
+                flags |= Document.FLAG_DIR_SUPPORTS_CREATE;
+            }
+        } else if (file.canWrite()) {
+            // If the file is writable set FLAG_SUPPORTS_WRITE and
+            // FLAG_SUPPORTS_DELETE
+            flags |= Document.FLAG_SUPPORTS_WRITE;
+            flags |= Document.FLAG_SUPPORTS_DELETE;
+
+            // Add SDK specific flags if appropriate
+            if (SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                flags |= Document.FLAG_SUPPORTS_RENAME;
+            }
+            if (SDK_INT >= Build.VERSION_CODES.N) {
+                flags |= Document.FLAG_SUPPORTS_REMOVE;
+                flags |= Document.FLAG_SUPPORTS_MOVE;
+                flags |= Document.FLAG_SUPPORTS_COPY;
+            }
+        }
+
+        final String displayName = file.getName();
+        final String mimeType = getTypeForFile(file);
+
+        if (mimeType.startsWith("image/")) {
+            // Allow the image to be represented by a thumbnail rather than an icon
+            flags |= Document.FLAG_SUPPORTS_THUMBNAIL;
+        }
+
+        final MatrixCursor.RowBuilder row = result.newRow();
+        row.add(Document.COLUMN_DOCUMENT_ID, docId);
+        if(file.getAbsolutePath() == mBaseDir.getAbsolutePath()) {
+            row.add(Document.COLUMN_DISPLAY_NAME, getContext().getApplicationInfo().loadLabel(getContext().getPackageManager()).toString());
+        } else {
+            row.add(Document.COLUMN_DISPLAY_NAME, displayName);
+        }
+        row.add(Document.COLUMN_SIZE, file.length());
+        row.add(Document.COLUMN_MIME_TYPE, mimeType);
+        row.add(Document.COLUMN_LAST_MODIFIED, file.lastModified());
+        row.add(Document.COLUMN_FLAGS, flags);
+
+        // Add a custom icon
+        row.add(Document.COLUMN_ICON, R.drawable.love);
```

Comment:
Maybe just custom LOVE icon for `application/x-love-game` MIME type?

Reply:
I'm not sure what you mean here?

Reply:
Instead of applying LOVE icon entirely to all files, apply it to .love files only.

Reply:
Ah alright I'll do that